pySDC/projects/PinTSimE/battery_model.py

```diff
@@ -245,23 +264,146 @@ def plot_voltages(description, problem, sweeper, use_switch_estimator, use_adapt
     plt_helper.plt.close(fig)
 
 
-def proof_assertions_description(description, problem_params):
+def check_solution(stats, problem, use_adaptivity, use_switch_estimator):
     """
-    Function to proof the assertions (function to get cleaner code)
+    Function that checks the solution based on a hardcoded reference solution. Based on check_solution function from @brownbaerchen.
+
+    Args:
+        stats (dict): Raw statistics from a controller run
+        problem (problem_class.__name__): the problem_class that is numerically solved
+        use_adaptivity (bool): flag if adaptivity wants to be used or not
+        use_switch_estimator (bool): flag if the switch estimator wants to be used or not
+    """
+
+    data = get_data_dict(stats, use_adaptivity, use_switch_estimator)
+
+    if use_switch_estimator and use_adaptivity:
+        if problem == 'battery':
+            msg = 'Error when using switch estimator and adaptivity for battery:'
+            expected = {
+                'cL': 0.5474500710994862,
+                'vC': 1.0019332967173764,
+                'dt': 0.011761752270047832,
+                'e_em': 8.001793672107738e-10,
+                'switches': 0.18232155791181945,
+                'restarts': 3.0,
+                'sum_niters': 44,
+            }
+        elif problem == 'battery_implicit':
+            msg = 'Error when using switch estimator and adaptivity for battery_implicit:'
+            expected = {
+                'cL': 0.5424577937840791,
+                'vC': 1.0001051105894005,
+                'dt': 0.01,
+                'e_em': 2.220446049250313e-16,
+                'switches': 0.1822923488448394,
+                'restarts': 6.0,
+                'sum_niters': 60,
+            }
+
+    got = {
+        'cL': data['cL'][-1],
+        'vC': data['vC'][-1],
+        'dt': data['dt'][-1],
+        'e_em': data['e_em'][-1],
+        'switches': data['switches'][-1],
+        'restarts': data['restarts'],
+        'sum_niters': data['sum_niters'],
+    }
+
+    for key in expected.keys():
+        assert np.isclose(
+            expected[key], got[key], rtol=1e-4
+        ), f'{msg} Expected {key}={expected[key]:.4e}, got {key}={got[key]:.4e}'
+
+
+def get_data_dict(stats, use_adaptivity=True, use_switch_estimator=True, recomputed=False):
```

Comment:
Maybe you can generalise this function to arbitrary number of switches. Then you can import it from `batter_2condensators_model.py`.

Reply:
Good idea!